examples/demo/visuals/wiggly_bar/wiggly_bar.py

```diff
@@ -0,0 +1,764 @@
+        for nameV, minV, maxV, typeV, iniV in self.param.parameters:
+            # Create Labels for each element
+            plist.append(QtGui.QLabel(nameV))
+
+            if nameV == 'x' or nameV == 'scale':
+                # 'x' is the start of the 'Initial Conditions' groupbox,
+                # 'scale' is the start of the 'Display Parameters' groupbox
+                important_positions.append(len(plist) - 1)
+
+            # Create Spinboxes based on type - doubles get a DoubleSpinBox,
+            # ints get regular SpinBox.
+            # Step sizes are the same for every parameter except font size.
+            if typeV == 'double':
+                self.psets.append(QtGui.QDoubleSpinBox())
+                self.psets[-1].setDecimals(3)
+                if nameV == 'font size':
+                    self.psets[-1].setSingleStep(1.0)
+                else:
+                    self.psets[-1].setSingleStep(0.01)
+            elif typeV == 'int':
+                self.psets.append(QtGui.QSpinBox())
+
+            # Set min, max, and initial values
+            self.psets[-1].setMaximum(maxV)
+            self.psets[-1].setMinimum(minV)
+            self.psets[-1].setValue(iniV)
+
+        pidx = -1
+        for pos in range(len(plist)):
+            if pos in important_positions:
+                pidx += 1
+            param_boxes_layout[pidx].addWidget(plist[pos], pos + pidx, 0)
+            param_boxes_layout[pidx].addWidget(self.psets[pos], pos + pidx, 1)
+            self.psets[pos].valueChanged.connect(self.update_parameters)
+
+        param_boxes_layout[0].addWidget(QtGui.QLabel('Method: '), 8, 0)
+        param_boxes_layout[0].addWidget(self.method_options, 8, 1)
+        param_boxes_layout[-1].addWidget(self.pivot_chk, 2, 0, 3, 0)
+
+        for groupbox, layout in zip(self.groupbox_list, param_boxes_layout):
+            groupbox.setLayout(layout)
+
+        for groupbox in self.groupbox_list:
+            self.splitter.addWidget(groupbox)
+
+        vbox = QtGui.QVBoxLayout()
+        hbox = QtGui.QHBoxLayout()
+        hbox.addWidget(self.splitter)
+        hbox.addStretch(5.0)
+        vbox.addLayout(hbox)
+        vbox.addStretch(1.0)
+
+        self.setLayout(vbox)
+
+    def update_parameters(self, option):
+        self.param.props['pivot'] = self.pivot_chk.isChecked()
+        self.param.props['method'] = self.method_list[
+            self.method_options.currentIndex()
+        ]
+        keys = map(lambda x: x[0], self.param.parameters)
+        for pos, nameV in enumerate(keys):
+            self.param.props[CONVERSION_DICT[nameV]] = self.psets[pos].value()
+        self.changed_parameter_sig.emit(self.param)
+
+
+class MainWindow(QtGui.QMainWindow):
+
+    def __init__(self, param=None):
+        """Main Window for holding the Vispy Canvas
+        and the parameter control menu"""
+        QtGui.QMainWindow.__init__(self)
+
+        self.resize(1067, 800)
+        self.setWindowIcon(QtGui.QIcon('spring.ico'))
```

Comment:
Ah, I'll move the .ico file there, then. Should I throw out the README, or leave that in as well?